src/common/verify/isValidAccessGrant.ts

```diff
@@ -48,7 +51,20 @@ async function isValidAccessGrant(
   } = {},
 ): Promise<{ checks: string[]; warnings: string[]; errors: string[] }> {
   const fetcher = await getSessionFetch(options);
-  const vcObject = await getBaseAccess(vc, options);
+  let validVC = null;
+  if (
+    typeof vc !== "string" &&
+    !isUrl(vc.toString()) &&
+    !(vc instanceof URL) &&
+    !isDatasetCore(vc)
+  ) {
+    validVC = await verifiableCredentialToDataset(vc, {
+      includeVcProperties: true,
+      requireId: false,
+    });
+  }
+  // validVc is null if the provided VC is of the supported types
+  const vcObject = await getBaseAccess(validVC ?? vc, options);
```

Comment:
```suggestion
  if(validVc === null && isUrl(vc.toString())) {
    validVc = vc;
  }
  if(validVc === null) {
    throw new Error(`Invalid argument: expected either a VC URL or a RDFJS DatasetCore, received ${vc}`)
  }
  
  const vcObject = await getBaseAccess(validVc, options);
```
This allows to add an explicit error being thrown in case we didn't know how to handle the provided argument.